confd/pkg/backends/calico/client.go

```diff
@@ -1570,3 +1613,34 @@ func withDefault(val, dflt string) string {
 	}
 	return dflt
 }
+
+func (c *client) updateNodeMeshPassword() {
+	// Retrieve the default BGPConfig for the password secret information
+	key := model.ResourceKey{
+		Kind:      apiv3.KindBGPConfiguration,
+		Name:      globalConfigName,
+		Namespace: "",
+	}
+
+	cfgKVPair, err := c.client.Get(context.Background(), key, "")
```

Comment:
Hm, this is a bit unusual. Why do we need to query the datastore here rather than use the value we've received over the syncer? 

IIIUC, this is going to add a query to the API every time BGP peers change, which isn't that terrible but still would be nice to avoid.

Reply:
In the case where the secret that is referenced by the BGP config is changed, we won't have an "update" for the BGP config come through but instead will just get an empty update with a flag for updating peers from the secret watcher. In this case, we still want to update the BGP config but we will not have the reference for it since the "update" sent by the secret watcher will be empty.

I wasn't particularly sure if we had a way to avoid the query unless I kept a copy of the global BGP config cached to the client (which wouldn't be the worst thing) but then I would be worried about some weird timing issues where the secret is updated really close to when the BGP config is changed to reference a different secret. Not sure if that's preferable?

Reply:
actually, after thinking over the cases again, I think caching the BGP config should be fine since the secret values are also updated when the BGP config itself is updated so I don't think we'll run into any issues caching instead of calling the API separately. Let me try that out.